radio/src/gui/common/stdlcd/draw_functions.cpp

```diff
@@ -575,3 +575,36 @@ void drawSourceValue(coord_t x, coord_t y, source_t source, LcdFlags flags)
   getvalue_t value = getValue(source);
   drawSourceCustomValue(x, y, source, value, flags);
 }
+
+void drawFatalErrorScreen(const char * message)
+{
+  lcdClear();
+  lcdDrawText((LCD_W - getTextWidth(message, 0, DBLSIZE)) / 2,
+              LCD_H/2 - FH, message, DBLSIZE);
+  WDG_RESET();
+  lcdRefresh();
+  lcdRefreshWait();
+}
+
+void runFatalErrorScreen(const char * message)
+{
+  while (true) {
+    backlightEnable();
+    drawFatalErrorScreen(message);
+
+    uint8_t refresh = false;
+    while (true) {
```

Comment:
Should this while loop not include WDG_RESET();?

Reply:
Or is this intentional to reboot radio after some time using WDG?

Reply:
Maybe I’d better check where I have that code from. Sound like a good idea, but it’s not done either at the original place. I should check why.